libs/go/libp2p_node/aea/api.go

```diff
@@ -360,10 +362,11 @@ func UnmarshalEnvelope(buf []byte) (*Envelope, error) {
 	return envelope, err
 }
 
-func (aea *AeaApi) listen_for_envelopes() {
+func (aea *AeaApi) listenForEnvelopes() {
 	//TOFIX(LR) add an exit strategy
 	for {
-		envel, err := read_envelope(aea.pipe)
+		envel, err := HandleAcnMessageFromPipe(aea.pipe, aea, aea.AeaAddress())
+
 		if err != nil {
```

Comment:
Not sure why we stop when there's an error...

Reply:
updated